sdk/keyvault/azure-security-keyvault-common/src/sha.cpp

```diff
@@ -59,7 +59,11 @@ class SHAWithOpenSSL final : public Azure::Core::Cryptography::Hash {
 public:
   SHAWithOpenSSL(SHASize size)
   {
+#if (OPENSSL_VERSION_MAJOR == 1 && OPENSSL_VERSION_MINOR == 0) || OPENSSL_VERSION_MAJOR == 0
+    if ((m_context = EVP_MD_CTX_create()) == NULL)
```

Comment:
Isn't OpenSSL 0.x and 1.0 already end of life?  

We should not allow, for instance TLS version lower than 1.3.  I think Azure services won't work with lower versions. I remember someone mentioned something about it once. Was it you @ahsonkhan ?

I would be happier asking for OpenSSL 1.1 as MIN req.

Reply:
Looks like Azure requires TLS 1.2 
https://azure.microsoft.com/en-us/updates/azuretls12/#:~:text=Microsoft%20Azure%20recommends%20all%20customers%20complete%20migration%20towards,a%20switch%20to%20accept%20only%20TLS%201.2%20traffic.

So, this would let versions 0.x out of the scope here and we should not compile those.

Reply:
But there's no harm to give our customers more choices. If they want to use an older version of OpenSSL, they do it at their own risk.

Reply:
Plus, this piece of code is just to calculate SHA-256 hash. it has nothing to do with TLS version.

If some Azure service decides to enforce minimum TLS version (say storage requires minimum TLS1.2), it should be done in service sdk.